markdown_it/tree.py

```diff
@@ -211,27 +211,22 @@ def _set_children_from_tokens(self, tokens: Sequence[Token]) -> None:
         while reversed_tokens:
             token = reversed_tokens.pop()
 
-            if token.nesting == 0:
-                child = self._make_child(token=token)
-                if token.children:
-                    child._set_children_from_tokens(token.children)
+            if not token.nesting:
+                self._add_child([token])
                 continue
-
-            assert token.nesting == 1
+            if token.nesting != 1:
+                raise ValueError("Invalid token nesting")
 
             nested_tokens = [token]
             nesting = 1
-            while reversed_tokens and nesting != 0:
+            while reversed_tokens and nesting:
                 token = reversed_tokens.pop()
                 nested_tokens.append(token)
                 nesting += token.nesting
-            if nesting != 0:
+            if nesting:
                 raise ValueError(f"unclosed tokens starting {nested_tokens[0]}")
```

Comment:
maybe same here as above for noting the index of the problematic token in the token stream